libs/shapes/Dandelion.cpp

```diff
@@ -0,0 +1,112 @@
+#include "Dandelion.h"
+#include "soo_vectors.h"
+
+namespace soo {
+
+Dandelion::Dandelion(ofVec2f bbox_top_left_corner, float bbox_edge_length) {
+  // Define the bouding box
+  bbox_ = ofRectangle(bbox_top_left_corner.x, bbox_top_left_corner.y,
+                      bbox_edge_length, bbox_edge_length);
+
+  // Define the circle a bit to the left and bottom of the center of the
+  // bounding box
+  circle_center_.x = bbox_top_left_corner.x + 0.45f * bbox_edge_length;
+  circle_center_.y = bbox_top_left_corner.y + 0.55f * bbox_edge_length;
+  circle_radius_ = 0.6f * bbox_edge_length / 2.f;
+
+  // Define the ellipse a bit to the left and bottom of the center of the circle
+  ellipse_center_.x = circle_center_.x - 0.2f * circle_radius_;
+  ellipse_center_.y = circle_center_.y + 0.2f * circle_radius_;
+  ellipse_width_ = 0.8f * circle_radius_;
+  ellipse_height_ = 0.5f * circle_radius_;
+
+  // Define the dandelion lines such that:
+  // - the origin of the line is set to the center of the ellipse
+  // - the direction of the line is random
+  // - the line length depends on where the direction is pointing to:
+  //   - 1st quadrant of the circle (+, -): longer x, longer y
+  //   - 2nd quadrant of the circle (-, -): shorter x, longer y
+  //   - 3rd quadrant of the circle (-, +): shorter x, shorter y
+  //   - 4th quadrant of the circle (+, +): longer x, shorter y
+  // - some of the lines should be arrows:
+  //   - up to 75% of the the longest lines can be set as arrows
+  //   - up to 10% of the rest of the lines can be set as arrows
+  unsigned long num_lines = static_cast<unsigned long>(bbox_edge_length / 5);
+  lines.reserve(num_lines);
+
+  for (unsigned long i{0}; i < num_lines; i++) {
+    dandelion::Line line;
+    line.origin_ = ellipse_center_;
+    line.direction_ = vectors::GetRandomUnitVec2();
+
+    const float x = line.direction_.x;
+    const float y = line.direction_.y;
+    constexpr float min_factor = 0.5f;
```

Comment:
const naming: kMinFactor

Reply:
or maybe better don't const any of the parameters at all, `x`, `y` nor `min_factor`